content/en/blog/_posts/2021-02-27-annotating-k8s-for-humans.md

```diff
@@ -0,0 +1,99 @@
+
+## Human Service Discovery
+
+Troubleshooting always begins with information gathering. While much attention has been paid to centralizing machine data (e.g., logs, metrics), much less attention has been given to the human aspect of service discovery. Who owns a particular service? What Slack channel does the team work on? Where is the source for the service? What issues are currently known and being tracked?
+
+## Kubernetes Annotations
+
+Kubernetes annotations are designed to solve exactly this problem. Oft-overlooked, Kubernetes annotations are designed to add metadata to Kubernetes objects. The Kubernetes documentation says annotations can “attach arbitrary non-identifying metadata to objects.” This means that annotations should be used for attaching metadata that is external to Kubernetes (i.e., metadata that Kubernetes won’t use to identify objects. As such, annotations can contain any type of data. This is a contrast to labels, which are designed for uses internal to Kubernetes. As such, label structure and values are [constrained](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) so they can be efficiently used by Kubernetes.
+
+
+## Kubernetes Annotations in Action
+
+Here is an example. Imagine you have a Kubernetes service for quoting, called the quote service. You can do the following:
+
+```
+$ kubectl annotate service quote a8r.io/owner=”@sally”
+```
+
+In this example, we’ve just added an annotation called `a8r.io/owner` with the value of @sally. Now, we can use `kubectl describe` to get the information.
+
+```
+$ kubectl describe svc quote
+Name:              quote
+Namespace:         default
+Labels:            <none>
+Annotations:       a8r.io/owner: @sally
+Selector:          app=quote
+Type:              ClusterIP
+IP:                10.109.142.131
+Port:              http  80/TCP
+TargetPort:        8080/TCP
+Endpoints:         <none>
+Session Affinity:  None
+Events:            <none>
+```
+
+If you’re practicing GitOps (and you should be!) you’ll want to code these values directly into your Kubernetes manifest, e.g.,
+
+```
+apiVersion: v1
+kind: Service
+metadata:
+  name: quote
+  annotations:
+    a8r.io/owner: “@sally”
+spec:
+  ports:
+  - name: http
+    port: 80
+    targetPort: 8080
+  selector:
+    app: quote
+```
+
+## A Convention for Annotations
+
+Adopting a common convention for annotations ensures consistency and understandability. Typically, you’ll want to attach the annotation to the service object, as services are the high-level resource that maps most clearly to a team’s responsibility. Namespacing your annotations is also very important. Here is one set of conventions, documented at [a8r.io](https://a8r.io), and reproduced below:
+
+| Annotation                                 | Description                                 |
```

Comment:
If you want, you can add a [table caption](https://kubernetes.io/docs/contribute/style/hugo-shortcodes/#table-captions) for accessibility.